src/ParseObject.js

```diff
@@ -2492,8 +2492,11 @@ const DefaultController = {
             return Promise.reject(objectError);
           }
           for (const object of target) {
-            await localDatastore._updateLocalIdForObject(mapIdForPin[object.id], object);
-            await localDatastore._updateObjectIfPinned(object);
+            // Make sure that it is a ParseObject before updating it into the localDataStore
+            if(object instanceof ParseObject) {
```

Comment:
Can you add a space after the if?
```
if (object instanceof ParseObject) {
```

There is another place where `_updateLocalIdForObject` is used can you add the check there and update the tests if needed

Reply:
I've edited to include the space.

Regarding the other place that references `_updateLocalIdForObject`, do you mean this?
https://github.com/parse-community/Parse-SDK-JS/blob/fce7f31f2db8293279ee77a6a2dd08d40e725c2a/src/ParseObject.js#L2531-L2532
If so it's ok to not check again I think, since `target instanceof ParseObject` has already been checked as part of its parent `if` condition.
https://github.com/parse-community/Parse-SDK-JS/blob/fce7f31f2db8293279ee77a6a2dd08d40e725c2a/src/ParseObject.js#L2504

Reply:
Ok an additional check isn't needed.